core/includes/theme.inc

```diff
@@ -1951,7 +1951,19 @@ function theme_dropbutton_wrapper($variables) {
  */
 function theme_image($variables) {
   $attributes = $variables['attributes'];
-  $attributes['src'] = file_create_url($variables['uri']);
+
+  // Check for missing 'uri', allow D7's 'path' instead.
+  $uri = $variables['uri'];
+  if (!$uri && !empty($variables['path'])) {
```

Comment:
If they set both `uri` and `path` to be cross-platform compatible, then Backdrop will use `uri` and doesn't care if `path` is still set.  No warning is needed, because `theme_image()` has everything it needs and expects.

It is only when `uri` is not set and `path` is set and Backdrop has to drop back to Drupal 7 compatibility mode that the message is needed.